### PR TITLE
Refactors training loop, removes bidirectional message passing between trainer and algo

### PR DESCRIPTION
**Pull request by lucasosouza**
Do not merge, please review first. Tests still required to ensure logging and training is the same

Also pending some clarifications from the metaworld/garage work group regarding the training flow in general and whether or not updating tasks every epoch is required in v2. 

One other issue, possibly very relevant: runtimes are higher. The change is likely in the underlying libraries, metaworld and/or garage, since this PR installs from master instead. It is impossible to tell whether these changes that led to higher runtime actually fixed something or not, so it is a risk (or it is the other way, that the branch we were pulling from prior to this PR is actually the latest version that fixes a bug, but hasn't been merged yet). On a p3.2x is taking 31 sec to update gradient vs 16 sec in the prior version.

### CODE REVIEW

File: nupic/embodied/multitask/algos/custom_mtsac.py [hunk 1]
```diff
@@ -92,121 +87,163 @@ def __init__(
...
+                samples = as_torch_dict(self.replay_buffer.sample_transitions(
+                    self._buffer_batch_size
+                ))
+                policy_loss, qf1_loss, qf2_loss = self.optimize_policy(samples)
+                total_losses[0] += policy_loss
+                total_losses[1] += qf1_loss
+                total_losses[2] += qf2_loss
+                self._update_targets()
+
+        return total_losses
```
Comment (abhi-iyer) [inline nupic/embodied/multitask/algos/custom_mtsac.py:193]: NIT: can we normalize losses here instead of outside this method?

**Comment by abhi-iyer** [comment]
Should we also add the log_per_task flag in the LoggingArguments section in the parser?

**Comment by abhi-iyer** [comment]
LGTM otherwise!

**Comment by lucasosouza** [comment]
Test run ongoing. If results are within the expected range, merging right after.
